### ISSUE TITLE
Stop training when the loss is not changing much

### ISSUE THREAD

**Issue by BJohn-dev** [flninja12/Memristor_learning_model#3]
the learning process should be stopped by sufficient basis not just by limiting. 
Infinite loop until loss doesn't moves a lot.

**Comment by BJohn-dev** [comment]
loss function is not effectively designed. Crossbar output current is to small(~0.2mA) so it should be scaled up to use softmax effectively.
